### PR TITLE
Add WooCommerce Services to Setup Wizard

### PR DESCRIPTION
**Pull request by nabsul**
Note: This is in progress, but ready for review - it just needs some styling tweaks.

Add WooCommerce Services to Setup Wizard 

- Rename "store locale" to "store location"
- Move tax setup to to store location page
- Adds dynamic rendering of the to-be-imported tax rates (based on location)
- Rename shipping & taxes to just shipping
- Add checkbox for installing WCS
- Uses same mechanism as payment gateway installer to conditionally install WCS

@jeffstieler @v18 @mikejolley @nabsul @olaolusoga

### CODE REVIEW

**Comment by olaolusoga** [comment]
Looking good!

screen shot 2017-05-29 at 9 59 14 pm

Can we:
- Change the checkbox to a radio button for "I will be..."?
- Align "Print a label, take advantage ..." with " Ship with ..."

screen shot 2017-05-29 at 10 15 30 pm

**Review by mikejolley — APPROVED** [review]
Code looks good to merge. I just have some styling and layout feedback I'll add separately.

**Comment by mikejolley** [comment]
On the shipping tab, you can lose the "default..." part of the label - there is no overriding this setting to it's not really a default. It's the actual unit used everywhere.

---

Since this is now the shipping step, ( @olaolusoga can judge if this is a bad idea ) we could actually  just have it default to "on" if you're proceeding, and "off" if you're skipping the step.

Then you only need the 2 unit settings and a single checkbox for services. The presentation of that could use some styling work and maybe a graphic? The styling of the theme step for storefront is much nicer.

---

Thats it 👍

**Comment by olaolusoga** [comment]
>Since this is now the shipping step, ( @olaolusoga can judge if this is a bad idea ) we could actually just have it default to "on" if you're proceeding, and "off" if you're skipping the step.
Then you only need the 2 unit settings and a single checkbox for services. The presentation of that could use some styling work and maybe a graphic? The styling of the theme step for storefront is much nicer.

Can you explain a little more @mikejolley ? Are you saying: Get rid of the "I will be shipping..." and "I will not be...", and just have an on/off for shipping as a whole, and  WCS will be the only checkbox on page?

**Comment by mikejolley** [comment]
@jeffstieler @nabsul @olaolusoga This would be my suggestion for presentation (https://github.com/woocommerce/woocommerce/tree/update/woo-services-presentation).

- Keep messaging brief - there is a learn more link if needed
- Matches gateway step layout
- Services is opt-in, not opt-out (sneaky sneaky), but recommended
- No option to turn off shipping because a user can just hit "skip". This has no negative impact on admin.
- The 'no shipping notice' can be turned off if services is enabled to avoid notice overload

services

**Comment by claudiulodro** [comment]
On this branch only, the Storefront message looks like this for me:

screen shot 2017-05-31 at 9 42 47 am

File: includes/admin/class-wc-admin-setup-wizard.php [hunk 1]
```diff
@@ -692,7 +713,7 @@ public function wc_setup_payments() {
 		?>
 		<h1><?php esc_html_e( 'Payments', 'woocommerce' ); ?></h1>
 		<form method="post" class="wc-wizard-payment-gateway-form">
-			<p><?php printf( __( 'WooCommerce can accept both online and offline payments. <a href="%1$s" target="_blank">Additional payment methods</a> can be installed later and managed from the <a href="%2$s" target="_blank">checkout settings</a> screen.', 'woocommerce' ), esc_url( admin_url( 'admin.php?page=wc-addons&view=payment-gateways' ) ), esc_url( admin_url( 'admin.php?page=wc-settings&tab=checkout' ) ) ); ?></p>
+			<p><?php printf( __( 'WooCommerce can accept both online and offline payments. <a href="%1$s" target="_blank">Additional payment methods</a> can be installed later and managed from the <a href="%2$s" target="_blank">checkout settings</a> screen.', 'woocommerce' ), esc_url( admin_url( 'admin.php?page=wc-addons&view=payment_gateways' ) ), esc_url( admin_url( 'admin.php?page=wc-settings&tab=checkout' ) ) ); ?></p>
```
Comment (claudiulodro) [inline includes/admin/class-wc-admin-setup-wizard.php]: I think the correct link should be `admin.php?page=wc-addons&section=payment_gateways`. `admin.php?page=wc-addons&view=payment_gateways` will drop you off at the featured extension screen.

**Comment by claudiulodro** [comment]
Looks good to me. Just needs one link updated then we're good to merge.

File: includes/admin/class-wc-admin-setup-wizard.php (hunk 1 above)
Comment (mikejolley) [inline includes/admin/class-wc-admin-setup-wizard.php]: Hmm this was payment_gateways on my install. Maybe it was the test file from https://github.com/woocommerce/woocommerce/pull/15368 breaking it..

Comment (claudiulodro) [inline includes/admin/class-wc-admin-setup-wizard.php]: [Looks to me like 'section' is used to determine the screen.](https://github.com/woocommerce/woocommerce/blob/master/includes/admin/class-wc-admin-addons.php#L373)

Comment (nabsul) [inline includes/admin/class-wc-admin-setup-wizard.php]: Hmmm... not sure how this change happened. Probably a rebase mistake on my part. I'll double check all the files for this type of thing.

Comment (mikejolley) [inline includes/admin/class-wc-admin-setup-wizard.php]: `View` may have been a longstanding bug but only just noticed :)

**Pull request merged**